Shopify/polaris-react
# Update to latest sewing-kit

### WHY are these changes introduced?

Keeping up to date, includes new linting rules mostly, meaning when eslint-plugin-shopify jumps to use the new typescript parser there will be less effort to update it and get onto a newer typescript version.

### WHAT is this pull request doing?
Update a few dependencies that we want to keep in sync with sewing-kit

Use `sewing-kit optimize` to optimize svgs instead of maintaining our
own script which does the same thing.

---

An update to babel-plugin-shopify has changed the default value of "useBuiltIns". We now explicitly set "useBuildIns: 'entry'" in our babel.config.js to maintain the same behaviour as previously. Not doing this results in adding ~50kb extra polyfill stuff.

This doesn't affect polaris when used as part of sewing-kit because sk doesn't use the stand-alone build.

### To Review

This PR is split into multiple commits, one of which is all the automatic fixes from running `yarn run format` which contains the majority of the code changes (apart from the yarn.lock file). It might be easier to look at the commits one-by-one rather than scrolling through a load of automatic fixes.

### To Test

* Ensure tests / linting / type checks pass
* Ensure aria labels still appear as you'd expect in the components where I had to add translations
